### PR TITLE
Chore: APP-2105 Refactor useDaoDetails

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [APP-2105](https://aragonassociation.atlassian.net/browse/APP-2105)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2105]: https://aragonassociation.atlassian.net/browse/APP-2105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/web-app/src/hooks/useDaoDetailsQuery.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ */
+export const useDaoQuery = (
+  daoAddressOrEns: string | undefined,
+  refetchInterval = 0
+) => {
+  const {network, networkUrlSegment} = useNetwork();
+  const {client, network: clientNetwork} = useClient();
+
+  // if network is unsupported this will be caught when compared to client
+  const queryNetwork = networkUrlSegment ?? network;
```
Comment (sepehr2github) [inline packages/web-app/src/hooks/useDaoDetailsQuery.tsx]: I think using useMemo here would be a small improvement to have ;D

File: packages/web-app/src/hooks/useDaoDetailsQuery.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    return fetchDaoDetails(client, daoAddressOrEns);
+  }, [client, daoAddressOrEns]);
+
+  return useQuery<DaoDetails | null>({
+    queryKey: ['daoDetails', daoAddressOrEns, queryNetwork],
+    queryFn,
+    select: addAvatarToDao,
+    enabled,
+    refetchOnWindowFocus: false,
+    refetchInterval: () => (refetchInterval ? refetchInterval : false),
```
Comment (sepehr2github) [inline packages/web-app/src/hooks/useDaoDetailsQuery.tsx]:
```suggestion
    refetchInterval: () => refetchInterval || false,
```

gpt suggested this not me xD

Comment (Fabricevladimir) [reply packages/web-app/src/hooks/useDaoDetailsQuery.tsx]: Actually I think I can just pass in `refetchInterval` if It's defaulted to 0 because it knows not to refetch when the value is 0

File: packages/web-app/src/context/network.tsx [hunk 3]
```diff
@@ -22,12 +22,14 @@ type NetworkContext = {
   network: SupportedNetworks;
   setNetwork: (network: SupportedNetworks) => void;
   isL2Network: boolean;
+  networkUrlSegment: string | undefined;
```
Comment (sepehr2github) [inline packages/web-app/src/context/network.tsx:25]: Could you explain this a bit? why did you add it here? to fix what issue?

File: packages/web-app/src/context/createDao.tsx [hunk 4]
```diff
@@ -59,9 +52,9 @@ const CreateDaoProvider: React.FC = ({children}) => {
   const {t} = useTranslation();
   const {getValues} = useFormContext<CreateDaoFormData>();
   const {client} = useClient();
-  const cachedDaoCreation = useReactiveVar(pendingDaoCreationVar);
-  const favoriteDaoCache = useReactiveVar(favoriteDaosVar);
-  const {preferences} = usePrivacyContext();
+
+  const addFavoriteDaoMutation = useAddFavoriteDaoMutation();
```
Comment (sepehr2github) [inline packages/web-app/src/context/createDao.tsx:56]: Well done, It's no less complex and more readable!

File: packages/web-app/src/context/network.tsx (hunk 3 above)
Comment (Fabricevladimir) [reply packages/web-app/src/context/network.tsx:25]: Because the network (state) is being set using `useEffect` and `useState`, there is an intermediate render where the network state is not the same as the network in the url. This means when we make a call to the SDK, we pass in the wrong network. That's why sometimes if you go to the Explorer page and connect your wallet to Goerli for example and click on a favorited DAO that's on Ethereum Mainnet, you will get a blank screen. What's happening is that before the network state gets updated to ETH Mainnet a call is made to the SDK looking for the DAO address on Goerli.

File: packages/web-app/src/hooks/useDaoDetailsQuery.tsx (hunk 1 above)
Comment (Fabricevladimir) [reply packages/web-app/src/hooks/useDaoDetailsQuery.tsx]: Barely. Getting to the point of diminishing returns I think lol

**Review by sepehr2github — APPROVED** [review]
Thanks for the changes
